service/src/main/java/io/camunda/service/GroupServices.java

```diff
@@ -118,8 +118,9 @@ public CompletableFuture<GroupRecord> updateGroup(final long groupKey, final Str
     return sendBrokerRequest(new BrokerGroupUpdateRequest(groupKey).setName(name));
   }
 
-  public CompletableFuture<GroupRecord> deleteGroup(final long groupKey) {
-    return sendBrokerRequest(new BrokerGroupDeleteRequest(groupKey));
+  public CompletableFuture<GroupRecord> deleteGroup(final String groupId) {
+    return sendBrokerRequest(
+        new BrokerGroupDeleteRequest(Long.parseLong(groupId)).setGroupId(groupId));
```

Comment:
💭 We gotta make sure we don't forget to remove this parsing and explicitly sending the command to partition 1 instead of basing it on the key. This reminds me we may have broken that for users/tenants/mappings as well 🤔

Reply:
I checked user, tenants and mapping. They are good.

Reply:
This is going to be part of this https://github.com/camunda/camunda/issues/29902 👍